### PR TITLE
working version of vercel app

### PR DESCRIPTION
**Pull request by walidvb**
The existing file did not work, nor did it correspond to the implemented vercel_bot

### CODE REVIEW

File: vercel-bot/api/index.ts [hunk 1]
```diff
@@ -3,6 +3,11 @@ import { Bot, webhookCallback } from 'grammy'
...
 const bot = new Bot(process.env['BOT_TOKEN'])
 
-export default async (req, res) => { // <-- Fill in the address of your deployed vercel allocation.
-  webhookCallback(bot, 'http')(req, res) // <-- The free version of vercel has restrictions on quotas, which we need to enable in the configuration file vercel.json
-}
\ No newline at end of file
+// attach all middleware
+bot.on("message", (ctx) => {
+  ctx.reply("Hi there!")
+})
```
Comment (KnorpelSenf) [inline vercel-bot/api/index.ts:9]:
```suggestion
bot.on("message", async (ctx) => {
  await ctx.reply("Hi there!")
})
```

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
LGTM, this is the only important change:

**Comment by KnorpelSenf** [comment]
Do you also want to adjust the README?

**Comment by walidvb** [comment]
@KnorpelSenf I can, but it still seems correct to me, not sure anything more is needed?
And the top README already mentions the vercel app, now 👌

File: vercel-bot/api/index.ts (hunk 1 above)
Comment (walidvb) [reply vercel-bot/api/index.ts:9]: I understood that the reply could not be read from a webhook? While more correct in terms of showing the acceptance of a promise, what's the point of waiting for the reply call to finish?
Beyond that, happy to make that change of course!

Comment (KnorpelSenf) [reply vercel-bot/api/index.ts:9]: > I understood that the reply could not be read from a webhook?

What do you mean?

> what's the point of waiting for the reply call to finish?

Lots and lots of bad things happen if you don't await your promises:
- You cannot do proper error handling. The errors will not always be shown in the logs. Instead, they just kill the instance.
- The message order is not guaranteed
- You may cause data loss if you use sessions
- You cannot access the result objects (not important in this case, but generally it is)
- probably more things that I cannot think of right now

File: vercel-bot/api/index.ts [hunk 2]
```diff
@@ -3,6 +3,11 @@ import { Bot, webhookCallback } from 'grammy'
...
-}
\ No newline at end of file
+// attach all middleware
+bot.on("message", async (ctx) => {
+  await ctx.reply("Hi there!")
+})
+
+// The free version of vercel has restrictions on quotas, which we need to enable in the configuration file vercel.json
+// webhookCallback will make sure that the correct middleware(listener) function is called
+export default webhookCallback(bot, 'express')(req, res)
```
Comment (KnorpelSenf) [inline vercel-bot/api/index.ts:13]: Sorry, I missed this.
```suggestion
export default webhookCallback(bot, 'express')
```

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]

File: vercel-bot/api/index.ts (hunk 2 above)
Comment (walidvb) [reply vercel-bot/api/index.ts]: 🤦

**Review by KnorpelSenf — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**